src/ucp/core/ucp_ep.c

```diff
@@ -2385,10 +2385,13 @@ int ucp_ep_config_test_rndv_support(const ucp_ep_config_t *config)
 
 void ucp_ep_do_keepalive(ucp_ep_h ep, ucp_lane_map_t *lane_map)
 {
-    ucp_lane_map_t check_lanes = *lane_map;
+    ucp_lane_map_t check_lanes;
     ucp_lane_index_t lane;
     ucs_status_t status;
 
+    /* Take updated ep_check_map, in case ep configuration has changed */
+    check_lanes = *lane_map & ucp_ep_config(ep)->key.ep_check_map;
```

Comment:
maybe update lane_mape in init_lanes?
or here? i.e. saving the result `*lane_map &= ucp_ep_config(ep)->key ep_check_map`?

Reply:
IMO it will be more intrusive to check if init_lanes ep is the current one and if yes then update keepalive.lane_map

Reply:
I meant to check it during init_lanes() and update keepalive.lane_map there
Is it possible?

Reply:
i think it would be more dangerous to update keepalive state this way, butter keepalive would not assume configuration stays the same. we could have several places that change the config.

Reply:
I see, thanks